### PR TITLE
Add junit reporter

### PR DESCRIPTION
**Pull request by mikoto2000**
See: #295, #296, #320

Reimplemented junit support.

Example in Azure DevOps:
image

### CODE REVIEW

**Review by etiennebarrie — APPROVED** [review]
Tested it and it worked great.

@wscourge Just to confirm, you're ok with no authorship for that?

**Comment by wscourge** [comment]
Yes, I'm ok

File: lib/erb_lint/reporters/junit_reporter.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+      def create_failure(filename, offense)
+        message = offense.message
+        type = offense.simple_name
+
+        failure_element = REXML::Element.new("failure", nil, CONTEXT)
+        failure_element.add_attribute("message", "#{type}: #{message}")
+        failure_element.add_attribute("type", type.to_s)
+
+        cdata_element = REXML::CData.new("#{filename}:#{offense.line_number}:#{offense.column}")
```
Comment (etiennebarrie) [inline lib/erb_lint/reporters/junit_reporter.rb:105]: Can we add more information here like the error message?

The way we build Buildkite annotations ignores the `message` attribute if there's text content in the element:

image

We could do a better job on our end to handle that, but it still feels like the text content could be more comprehensive.

#296 was using the message in the text of the element: https://github.com/Shopify/erb-lint/pull/296/files#diff-77e0e3207b9a621523ef25749ac60fce67a306eeb27715133264667ffbf2cbd4R14
ci-queue also has the more information than the location, the message and backtrace cleaned up a little bit; https://github.com/Shopify/ci-queue/blob/941e4fe39a5bfdfc3bb8c0ec2c5a0dfa71075c3e/ruby/lib/minitest/queue/junit_reporter.rb#L118

**Review by etiennebarrie — COMMENTED** [review]
Sorry just realized the output isn't as comprehensive as I'd like. Can we add more to the `<failure>` element text content?

File: lib/erb_lint/reporters/junit_reporter.rb (hunk 1 above)
Comment (mikoto2000) [reply lib/erb_lint/reporters/junit_reporter.rb:105]: Ok, I will fix the failure tag's text content to contain the message attribute.

Example: `app/views/subscriptions/_loader.html.erb:1:7` to `SpaceInHtmlTag: Extra space detected where there should be no space. at app/views/subscriptions/_loader.html.erb:1:7`.

Comment (mikoto2000) [reply lib/erb_lint/reporters/junit_reporter.rb]: @etiennebarrie 
I fixed by commit: 2fe02bb88860680c9a60c8785f40fe1a87876c5b.

Comment (mikoto2000) [reply lib/erb_lint/reporters/junit_reporter.rb]: I forgot fix test...
please wait.

Comment (mikoto2000) [reply lib/erb_lint/reporters/junit_reporter.rb]: @etiennebarrie 
Sorry, I re-committed: 9ef15e20da0ad46077c88b73bac06e4edd15d2c2

**Pull request merged**